xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/macros.vm

```diff
@@ -768,16 +768,21 @@ $fn
 #macro(panelheader $title)
 ## Note: We pass the Panel name as an HTML class attribute so that it's possible to style
 ## the Panels selectively using CSS.
-#set ($specialClassAttribute = '')
-#if ($paneldoc)
-  #set ($specialClassAttribute = $util.convertToAlphaNumeric($paneldoc.name))
+#set ($defaultDocumentName = $services.model.getEntityReference('DOCUMENT', 'default').name)
+#if ($paneldoc.documentReference.name == $defaultDocumentName)
+  #set ($serializedDocument = $!services.model.serialize($paneldoc.documentReference.parent, 'compactwiki'))
+#else
+  #set ($serializedDocument = $!services.model.serialize($paneldoc.documentReference, 'compactwiki'))
 #end
+#set ($specialClassAttribute = $stringtool.toAlphaNumeric("$!serializedDocument"))
+## We keep this class for compatibility purpose but the previous one should be used instead.
+#set ($deprecatedClassAttribute = $stringtool.toAlphaNumeric("$!paneldoc.name"))
```

Comment:
I would move the above lines in a macro to avoid the duplication. Also, if ``toAlphaNumeric`` handles ``null`` nicely (as it should, by returning null) then you probably don't need the silent notation (as long as you use the silent notation below).

Reply:
Sure it can be refactored.